pkg/kubectl/testing/fake.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/discovery"
+	fakedisco "k8s.io/client-go/discovery/fake"
+	"k8s.io/client-go/dynamic"
+	fakerest "k8s.io/client-go/rest/fake"
+	"k8s.io/client-go/scale"
+	testcore "k8s.io/client-go/testing"
+)
+
+func FakeScaleClient(discoveryResources []*metav1.APIResourceList, pathsResources map[string]runtime.Object, pathsOnError map[string]map[string]kerrors.StatusError) (scale.ScalesGetter, error) {
```

Comment:
yeah, I can hear you :) How bad this actually is ?

FYI: https://github.com/kubernetes/kubernetes/issues/56865

Reply:
> yeah, I can hear you :) How bad this actually is ?

@DirectXMan12 We really need a fake scale client we can use.  Is it just a matter of finding time?

Reply:
Yeah, IIRC, I have one that was usuable with a fake rest mapper, but it's a bit clunky.  Let me try and take some time today and make it better.

Reply:
https://github.com/p0lyn0mial/kubernetes/pull/2
^---------- the current fake scale client works fine for this

Reply:
that should also close #56865

Reply:
@DirectXMan12 thanks for showing me how to use the fake client :) I think it will work and we can remove this custom code.